.github/workflows/cifuzz.yml

```
@@ -0,0 +1,47 @@
+name: CIFuzz
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+ Fuzzing:
+   runs-on: ubuntu-latest
+   permissions:
+     security-events: write
+   steps:
+   - name: Build Fuzzers
+     id: build
+     uses: google/oss-fuzz/infra/cifuzz/actions/build_fuzzers@master
+     with:
+       oss-fuzz-project-name: ''
+       language: go
+
+   - name: Run Fuzzers
+     uses: google/oss-fuzz/infra/cifuzz/actions/run_fuzzers@master
+     with:
+       oss-fuzz-project-name: '${{ github.event.repository.name }}'
+       language: go
+       fuzz-seconds: 600
+       output-sarif: true
+
+   - name: Upload Crash
+     uses: actions/upload-artifact@v3
```

Recommend pinning to a release commit SHA
Ref: [Pinned Dependencies](https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies)

```suggestion
     uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
```